spm_functions.py

```diff
@@ -57,47 +71,144 @@ class Species:
+        self.ind_track = self.name.index(Species_Tracked)
+        self.ind_ion = self.name.index(ion)
+        # I no longer force this to be postive. The coeff of th ion is automatically set, but the 
+        #   value for n if determed by the user since I do not include the elctron as a species in the reaction
+        self.nuA_nF = self.nu[self.ind_ion]*A_s/n/F
 
+def residual(_,SV,i_ext,Anode,Cathode):
+    '''
+    Derivations (a=anode,s=sperator,c=cathode)
+
+    Change in Double Layer potential Anode [0]:
+    Eta_a = Phi_a - Phi_el_a - U_a ; Phi_an = 0 ; Phi_a - Phi_el_a = delta_Phi_dl_a
+    Eta_a = delta_Phi_dl_a - U_a => sub into Butler Volmer
+    i_ext/A_sg = i_dl_a + i_far_a ; -i_dl_a = Cap_dl_a*(d Delta_Phi_dl_a/ dt) 
+    (-i_far_a + i_ext/A_sg)/Cap_dl_a = d Delta_Phi_dl_a/dt
+
+    Change in Lithium concentration in the Anode [1]:
+    dN_Li/dt = -s_dot_Li+*A_surf*N_p ; dC_Li/dt = (dN_Li/dt)/(Vol*N_p) ; s_dot_Li+ = -i_far*nu_Li+/(n*F) ; A_suf/Vol = A_s
+    dC_Li/dt = i_far*nu_Li+*A_s/(n*F)
+    
+    Change in Double Layer potential Cathode [2]: 
+    Eta_c = Phi_c - Phi_el_c - U_c ; Phi_c - Phi_el_c = delta_Phi_dl_c 
+    Eta_c = delta_Phi_dl_a - U_c => sub into Butler Volmer
+    -i_ext/A_sg = i_dl_c + i_far_c ; i_dl_c = Cap_dl_c*(d Delta_Phi_c_dl/ dt) 
+    (-i_far_c - i_ext/A_sg)/Cap_dl_c = d Delta_Phi_c_dl/dt
+    
+    Change in Lithium concentration in the Cathode [3]:
+    dC_Li/dt = i_far*nu_Li+*A_s/(n*F)
+    '''
+    # The convention for double layer potential is electrode minus electrolyte
+    # For this reaction, Li+ and the electron are both on the same side of the reaction,
+    #   so the signs of their coefficients will always be the same which means nuA_nF
+    #   is positive. As a result, a positive i_far will cause the concentration
+    #   of lithium in the electrode to increase
+    # I hard code in the positions for the empty electrode in this reaction. Since I use the reaction
+    #   Lithiated_Electrode -> Li+ + Electrode + e- for both half cells, the empty elcetrode is the last partcipant in 
+    #   the list in both HCs
+    # I adjust the exchange current densities for concentration 
+    # I use mole fractions in place of activites when adjusting for concentration with the Open Cell Potential
+     
+    # Anode
+    V_a = SV[0]
+    
+    X_Li_a = SV[1]/Anode.C_int[Anode.ind_track] # effective molar concentration of Lithium (LiC6) in the anode [-]
+    Anode.activity[Anode.ind_track] = Anode.gamma[Anode.ind_track]*(X_Li_a) # update activity of the LiC6
+    Anode.activity[-1] = Anode.gamma[-1]*(1 - X_Li_a) # update activity of the C6
+    # The activity of the Li+ in the electroltye does not change because I am assuming the concentration is constant
+    
+    # Adjust exchange current density for concentration
+    #   uses effective molar concentration which is activity/activity_coefficient 
+    i_o_an = ((X_Li_a)**Anode.Beta)*(
```

Comment:
A couple notes:
This expression should be the product of the species activity concentrations, each raised to: 
- The forward stoichiometric coefficient times (1-beta) for reactants or:
- The reverse stoichiometric coefficient times beta, for products.
That is mostly what you have here, except: 

1. You divide activity by gamma, on line 186, and 
2. You need to also have (1-X_Li_a) in your product, to account for the vacancy concentration.

There is a slightly more general way to frame it, where activity concentrations are divided by reference concentrations, but it doesn't actually impact the results, here, so is not worth getting into (and is easier in a formatted writeup, rather than in GitHub comments ;)